packages/components/src/components/expandable/expandable.ts

```diff
@@ -0,0 +1,240 @@
+
+  /**
+   * Used to check whether the component is expanded or not.
+   */
+  @property({ type: Boolean, reflect: true }) open = false;
+
+  /** Inverts the expandable and sets the color of the gradient to primary. */
+  @property({ type: Boolean, reflect: true }) inverted = false;
+
+  /** Sets the height of the gradient to 24px for paragraph and 32px for leadtext. */
+  @property({ reflect: true }) variant: 'paragraph' | 'leadtext' = 'paragraph';
+
+  public localize = new LocalizeController(this);
+
+  private updateMaxHeight() {
+    const scrollHeight = this.contentPreview?.scrollHeight.toString();
+    this.style.setProperty('--max-height-pixel', `${scrollHeight}px`);
+    this.style.setProperty('--max-height', scrollHeight);
+  }
+
+  private onToggleClick() {
+    this.updateMaxHeight();
+    this.open = !this.open;
+  }
+
+  @watch('open', { waitUntilFirstUpdate: true })
+  onOpenChange() {
+    if (this.open) {
+      this.emit('sd-show');
+      this.updateComplete.then(() => {
+        this.emit('sd-after-show');
+      });
+    } else {
+      this.emit('sd-hide');
+      this.updateComplete.then(() => {
+        this.emit('sd-after-hide');
+      });
+    }
+
+    this.details.setAttribute('open', this.open.toString());
+  }
+
+  /** Opens the expandable */
+  async show() {
+    if (this.open) {
+      return undefined;
+    }
+
+    this.onToggleClick();
+    return waitForEvent(this, 'sd-after-show');
+  }
+
+  /** Closes the expandable */
+  async hide() {
+    if (!this.open) {
+      return undefined;
+    }
+
+    this.onToggleClick();
+    return waitForEvent(this, 'sd-after-hide');
+  }
+
+  render() {
+    return html`
+      <details>
+        <summary aria-hidden="true" class="cursor-pointer overflow-hidden list-none">
+          <slot name="clone"></slot>
+        </summary>
+        <div part="content" class="content content-preview overflow-hidden relative">
+          <slot></slot>
+        </div>
+      </details>
+      <button
+        part="toggle"
+        class=${cx(
+          'toggle focus-visible:outline-offset-0 block m-0 p-0 w-full overflow-visible bg-transparent text-current font-inherit leading-inherit align-middle cursor-pointer',
+          !this.inverted ? 'focus-visible:focus-outline' : 'focus-visible:focus-outline-inverted'
+        )}
+        @click=${this.onToggleClick}
+      >
+        ${this.open
+          ? html`
+              <slot name="toggle-open">
+                <div class="default-content flex items-center justify-center w-full">
+                  <sd-link
+                    size="lg"
+                    href="#"
+                    onclick="return false;"
+                    ?inverted=${this.inverted}
+                    tabindex="-1"
+                    standalone
+                    >${this.localize.term('showLess')}
+                    <sd-icon library="system" name="chevron-up" slot="icon-left"></sd-icon
+                  ></sd-link>
```

Comment:
Can't we use sd-interactive here? This feels weird to use sd-link but removing all interaction...